src/utils/validation.ts

```diff
@@ -3,18 +3,36 @@ import { exit } from "process";
 
 import { BLOCK_SETTINGS_FILE } from "../constants";
 import { formatName, logError, readBlockSettingsFile } from "./index";
+import { getCategoryNames } from "./network";
 
 const isCategoryValid = (category: string, categories: string[]): boolean => {
     return categories
         .map(cat => cat.toLowerCase())
         .includes(category.toLowerCase());
 };
 
-export const validateInputs = (
+export const validateCategory = async (
+    category: string,
+    categories: string[] = []
+): Promise<void> => {
+    const categoryNames =
+        categories.length > 0 ? categories : await getCategoryNames();
+
+    if (!isCategoryValid(category, categoryNames || [])) {
+        logError(
+            `${category} is not a valid category name. Please enter a valid category name:\n\t- ${(
+                categoryNames || []
+            ).join("\n\t- ")}`
```

Comment:
You could pull the code for generating the valid names out to a function for a bit more readability:

```
`${category} is not a valid category name. Please enter a valid category name:\n\t- ${validCategoryNames()}`
```